Applications/WebPortal/WebApp/library/source/components/button.tsx

```diff
@@ -0,0 +1,102 @@
+import { css, StyleSheet } from 'aphrodite/no-important';
+import * as React from 'react';
+
+interface Properties {
+
+  /** The text to display on the button. */
+  label: string;
+
+  /** The class name of the input box. */
+  className?: string;
+
+  /** Determines if the component is readonly. */
+  readonly?: boolean;
+
+  /** Additional CSS styles. */
+  style?: any;
+
+  /** Called when the value changes. */
+  onClick?: () => void;
+}
+
+/** A custom button component. */
+export class Button extends React.Component<Properties> {
+  public static readonly defaultProps = {
+    onClick: () => {}
+  }
+
+  public render() {
+    const style = (() => {
+      if(this.props.readonly) {
+        return Button.STYLE.readonly;
+      } else {
+        return Button.STYLE.default;
+      }
+    })();
+    return (
+      <button style={{...style, ...this.props.style}}
+          disabled={this.props.readonly}
+          className={css(Button.EXTRA_STYLE.button) + ' ' +
```

Comment:
Is there a way to specify multiple classnames without appending two strings together? Can you use an array of strings? What options does React provide?

Reply:
React doesn't have a baked in option. The closest option to using an array is `[class1, class2].join(' ')`

It's certainly more readable at least.